src/generic/DirectSum.jl

```diff
@@ -44,8 +44,13 @@ summands(M::DirectSumModule{T}) where T <: RingElement = M.m
 ###############################################################################
 
 function show(io::IO, N::DirectSumModule{T}) where T <: RingElement
-   print(io, "DirectSumModule over ")
-   print(IOContext(io, :compact => true), base_ring(N))
+   if get(io, :supercompact, false)
+     io = pretty(io)
+     print(io, Uppercase(), "DirectSumModule")
```

Comment:
I guess `Uppercase()` is used here to disable any previous use of `Lowecase()` from recursive calls, and that may answer one of my questions above... and that's because `directSumModule` is not what we want?